plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/handler/NewApplicationHandler.java

```diff
@@ -31,10 +34,15 @@
 
 	@Override
 	public Object execute(ExecutionEvent event) throws ExecutionException {
-		IProject project = UIUtils.getFirstElement(HandlerUtil.getCurrentSelection(event), IProject.class);
-		NewApplicationWizardModel model = new NewApplicationWizardModel(project);
-		NewApplicationWizard wizard = new NewApplicationWizard(model);
-		WizardUtils.openWizard(wizard, HandlerUtil.getActiveShell(event));
+		try {
+			IProject project = UIUtils.getFirstElement(HandlerUtil.getCurrentSelection(event), IProject.class);
+			Connection connection = ConnectionsRegistryUtil.getConnectionFor(project);
+			NewApplicationWizardModel model = new NewApplicationWizardModel(project, connection.getResourceFactory());
+			NewApplicationWizard wizard = new NewApplicationWizard(model);
+			WizardUtils.openWizard(wizard, HandlerUtil.getActiveShell(event));
+		}catch(NullPointerException e) {
```

Comment:
hmm, but in general terms throwing indicates an error whereas we simply didnt find a connection for a given project which isnt an error? why not simply "if not null then"